lib/galaxy/jobs/__init__.py

```diff
@@ -712,7 +712,9 @@ def __init__(self, job, queue, use_persisted_destination=False):
         # and job recovery fail.
         # Create the working dir if necessary
         self._create_working_directory()
-        self.dataset_path_rewriter = self._job_dataset_path_rewriter(self.working_directory)
+        # the path rewriter needs destination params, so it cannot be set up until after the destination has been
+        # resolved
+        self.__dataset_path_rewriter = None
```

Comment:
Since this attribute is used in the `TaskWrapper` should class you should use a single leading underscore, because the double leading underscore invokes name mangling. (Probably better to never using the double leading underscore unless really necessary).

Reply:
It'd work if I initialized it in `TaskWrapper`'s init... I had considered name mangling but it actually doesn't matter, since `TaskWrapper`'s `__dataset_path_rewriter` is intentionally not the same as `JobWrapper`'s.